File: LuckyNo.py
class Solution:
    def luckyNumbers (self, matrix: List[List[int]]) -> List[int]:
        res = []
        rows = len(matrix)
        cols = len(matrix[0])

        # Find the minimum value in each row and keep track of their column indices
        row_mins = [0] * rows
        min_cols = [0] * rows
        for i in range(rows):
            min_val = matrix[i][0]
            min_col = 0
            for j in range(1, cols):
                if matrix[i][j] < min_val:
                    min_val = matrix[i][j]
                    min_col = j
            row_mins[i] = min_val
            min_cols[i] = min_col

        # Check if the row minimums are also the maximum in their respective columns
        for i in range(rows):
            min_val = row_mins[i]
            min_col = min_cols[i]
            is_lucky = True
            for k in range(rows):
                if matrix[k][min_col] > min_val:
                    is_lucky = False
                    break
            if is_lucky:
                res.append(min_val)

        return res

File: LuckyNo.cpp
class Solution {
public:
    vector<int> luckyNumbers (vector<vector<int>>& matrix) {
        vector<int> res;
        int rows = matrix.size();
        int cols = matrix[0].size();
        
        // Find the minimum value in each row and keep track of their column indices
        vector<int> rowMins(rows);
        vector<int> minCols(rows);
        
        for (int i = 0; i < rows; ++i) {
            int minVal = matrix[i][0];
            int minCol = 0;
            for (int j = 1; j < cols; ++j) {
                if (matrix[i][j] < minVal) {
                    minVal = matrix[i][j];
                    minCol = j;
                }
            }
            rowMins[i] = minVal;
            minCols[i] = minCol;
        }
        
        // Check if the row minimums are also the maximum in their respective columns
        for (int i = 0; i < rows; ++i) {
            int minVal = rowMins[i];
            int minCol = minCols[i];
            bool isLucky = true;
            for (int k = 0; k < rows; ++k) {
                if (matrix[k][minCol] > minVal) {
                    isLucky = false;
                    break;
                }
            }
            if (isLucky) {
                res.push_back(minVal);
            }
        }
        
        return res;
    }
};
